v2/contribute/decisions/accountlinking/0005-allowing-email-verification-in-post-login-account-linking.mdx

```diff
@@ -0,0 +1,66 @@
+
+<DecisionHeader status="accepted" lastUpdate="2022-12-01" created="2022-11-30" deciders={["rishabhpoddar", "bhumilsarvaiya"]} proposedBy={["bhumilsarvaiya"]} />
+
+## Context and Problem Statement
+
+Should we allow email verification to get triggered during post login account or not? Currently, if the account info of new account (which can already exists from before or a new account needs to be created for the given account info) is not verified and should require verification is compulsory, we are sending a account link failure error. Instead, should we create a new recipe account if it does not already exists and trigger email verification flow? The problem can be explained better with an example here:
+- User is currently logged in to there account with email A (which has an associated primaryUserId)
+- User wants to link account with email B via email password flow.
+- Post login account linking API is called with email A's session and email B (and password )as new account input
+- Now because email B for email password recipe can only be verified once the recipe user is created, currently it's in unverified state as no recipe user even exists with email B. Also, we are not allowing recipe user account creation in this scenario.
+- Based on current flow, it would be impossible to allow linking of B to A unless email password account for B is already created and email B is verified.
+- To counter problem statement, we should
+    - allow creating recipe user even if it is unverified (and there doesn't exists any other account with email B that has a primaryUserId)
+    - trigger email verification flow for the recipe user with email B. The recipe user may be in existance prior to this API is called or it may be a new account to be created. In both scenarios, we would like to trigger a email verfication flow
+
+## Considered Options
+
+* allow account creation if necessary and trigger email verification flow
+
+
+## Decision Outcome
+
+Chosen option: allow account creation if necessary and trigger email verification flow, because
+
+* We also are allowing email verification flow to happen in the post login account linking API when the “new” account to link used to exist from before. The reason this is OK, is that in order for the new account to be linked in this API, the user has to login with that new account anyway whilst being logged in to their existing account. In order to login to the new account, they have to either have access to the social account, or the email ID, or know the password of that account.
+
+* Earlier we just threw an error saying not verified, which did not serve any purpose. We did this cause, we thought of some attack vectors example:
+    - Secenario 1:
+        - Attacker creates an account with email A, which belongs to them and a primaryUserId gets created
+        - Attacker tries to create an account with email B which doesn't belong to them will send a verification mail to actual user.
```

Comment:
```suggestion
        - Attacker then calls post login account linking API with email B which doesn't belong to them. This will send a verification mail to actual user.
```